pkg/apis/pipeline/v1beta1/taskref_conversion.go

```diff
@@ -0,0 +1,27 @@
+package v1beta1
+
+import (
+	"context"
+
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+)
+
+func (tr TaskRef) convertTo(ctx context.Context, sink *v1.TaskRef) {
+	sink.Name = tr.Name
+	sink.Kind = v1.TaskKind(tr.Kind)
+	sink.APIVersion = tr.APIVersion
+	// TODO: handle bundle
```

Comment:
could you please link an issue here and below? #4546 is probably the best one; you can link it for the resources work as well. This PR doesn't need to block on mine.